subsys/net/ip/net_shell.c

```diff
@@ -4036,6 +4057,36 @@ static int cmd_net_tcp_send(const struct shell *shell, size_t argc,
 	return 0;
 }
 
+static int cmd_net_tcp_recv(const struct shell *shell, size_t argc,
+			    char *argv[])
+{
+#if defined(CONFIG_NET_TCP)
+	int ret;
+#endif
+
+	if (shell_help_requested(shell)) {
+		shell_help_print(shell, NULL, 0);
+		return -ENOEXEC;
+	}
+
+#if defined(CONFIG_NET_TCP)
+	if (!tcp_ctx || !net_context_is_used(tcp_ctx)) {
+		PR_WARNING("Not connected\n");
+		return -ENOEXEC;
+	}
+
+	ret =  net_context_recv(tcp_ctx, tcp_recv_cb, K_FOREVER, NULL);
```

Comment:
IMHO, K_FOREVER is a bit dangerous. How about accepting a timeout arg? if you really like, can be made optional, defaulting say to 5s.

Reply:
sure